drv/cosmo-hf/src/main.rs

```diff
@@ -94,7 +94,15 @@ mod reg {
 
     pub const NOR: *mut u32 = BASE.wrapping_add(0x40);
     pub const SPICR: *mut u32 = NOR.wrapping_add(0x0);
+    pub mod spisr {
+        pub const RX_EMPTY_BIT: u32 = 1 << 6;
+    }
     pub const SPISR: *mut u32 = NOR.wrapping_add(0x1);
+    pub mod spicr {
+        pub const SP5_OWNS_FLASH: u32 = 1 << 31;
+        pub const RX_FIFO_RESET: u32 = 1 << 15;
+        pub const TX_FIFO_RESET: u32 = 1 << 7;
```

Comment:
I don't suppose there's an easily referenceable citation for where these bits come from we could point to in a comment?

Reply:
so "yes" and future work required here. Over in fpga-land our register specs are defined and as part of the build we generate these: https://github.com/oxidecomputer/hubris/blob/master/drv/grapefruit-seq-server/spi_nor_reg_map.json which are carried in here or you can read the .html memory map at that same location.

For gimlet/sidecar, these had minimal re-usable parts so a flat memory map for the whole fpga worked fine, and there is xtask code to transform this json into something usable in rust. As we expand our use of FPGAs I think this isn't the right path, we'd ideally like to have these blocks collocated with the drivers. I'm now generating these .json files on a per-module basis but some further discussion needs to happen to better transform these into usable rust structures.  I'm happy to skip the json step and generate rust directly also but getting some examples of what we want this to look like and how to generate it in a friendly, idiomatic way would be helpful and are really needed to move this forward.

For the minimal drivers we're creating here, the decision was to punt on the fancy generation work for expediency I think but I'm certainly interested in getting to a better place here long-term and will put work in on our side to make that happen if we have a strategy and direction!